### PR TITLE
Add fetch() funcitonality to say a message

### PR DESCRIPTION
**Pull request by tin0d0**

### CODE REVIEW

File: portfolio/src/main/webapp/script.js [hunk 1]
```diff
@@ -68,4 +68,14 @@ function open_neptune_modal() {
...
+}
+
+/**
+ * using async to fetch /data sections
+ * improves readability and simpleness
+ */
+async function getDataAsyncAwait() {
+  const response = await fetch('/data');
+  const comments = await response.text();
+  document.getElementById('data-container').innerText = comments;
```
Comment (frankpzn) [inline portfolio/src/main/webapp/script.js:80]: Probably it's cleaner to use then()

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 2]
```diff
@@ -26,7 +27,29 @@
...
+      * Converts a data array instance into a JSON string using manual String concatentation.
+    */
+  private String convertToJson(ArrayList<String> array) {
+    String json = "";
+    json += "\"" + array.get(0) + "\"";
+    json += ", ";
+    json += "\"" + array.get(1) + "\"";
+    json += ", ";
+    json += array.get(2);
+    return json;
```
Comment (frankpzn) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:52]: use String format instead.

Comment (frankpzn) [reply portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:52]: Also this assumes the array has 3 elements. Use for loop will be another option

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 3]
```diff
@@ -26,7 +27,29 @@
 
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    response.setContentType("text/html;");
-    response.getWriter().println("<h1>Hello Tino!</h1>");
+    ArrayList<String> comments = new ArrayList<String>();
+    comments.add("Hi tino");
+    comments.add("i like pie");
+    comments.add("nice space");
```
Comment (frankpzn) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:33]: It's best practice to define constants like these as static variables at the top.

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java (hunk 2 above)
Comment (HidyHan) [reply portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:52]: Please use Gson library for this: https://stackoverflow.com/questions/14228912/how-to-convert-list-to-json-in-java.

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java (hunk 3 above)
Comment (tin0d0) [reply portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:33]: Sounds good! Removed constants, and I'm now using GSON! Thank you!

**Pull request merged**